core/network/cidr.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"net"
+	"net/netip"
+	"strconv"
+
+	"github.com/juju/juju/internal/errors"
+)
+
+// CIDRAddressType returns back an AddressType to indicate whether the supplied
+// CIDR corresponds to an IPV4 or IPV6 range. An error will be returned if a
+// non-valid CIDR is provided.
+//
+// Caveat: if the provided CIDR corresponds to an IPV6 range with a 4in6
```

Comment:
I think this comment should be removed. It is silly and doesn't understand 4in6. If the address starts with `::ffff`, Then it is a v4 address and we should classify it as such. IMO the code is correct and this can just be removed.

Reply:
Ok, I just copied this from the other file. It's quite old code I think.